jbpm-benchmarks/jbpm-performance-tests-jmh/src/main/java/org/jbpm/test/performance/scenario/load/services/QueryProcessesAndTasksByVariables.java

```diff
@@ -127,32 +123,28 @@ public void init() throws Exception {
 
         // Update task variables twice
         taskVariables.forEach( (k, v) -> taskVariables.replace(k, counter.getAndIncrement()));
-        processIds.forEach( id -> {
-            runtimeDataService.getTasksByProcessInstanceId(id).forEach( taskId -> {
-                UserTaskInstanceDesc task = runtimeDataService.getTaskById(taskId);
-                Map<String, Object> newInputTask = new HashMap<>();
-                userTaskService.getTaskInputContentByTaskId(taskId).keySet().stream()
-                        .filter(o -> o.contains("task_var_in"))
-                        .forEach( key -> {
-                            newInputTask.put(key, taskVariables.get(key));
-                            userTaskService.updateTask(taskId, "salaboy", task, newInputTask, null);
-                        });
-            });
-        });
+        runtimeDataService.getTasksAssignedAsPotentialOwner("salaboy", new QueryFilter())
+                .forEach( taskSummary -> {
+                    Map<String, Object> newOutputTask = new HashMap<>();
+                    definitionService.getTaskOutputMappings(deploymentUnit.getIdentifier(), processDefinitionId, taskSummary.getName()).keySet().stream()
+                            .filter(s -> taskVariables.containsKey(s))
```

Comment:
Yes, that's correct. Each task has a different subset of input & output vars with different values between them. I've done it this way to avoid retrieving a huge number of records when using the new advanced query api and to make a little more realistic use case somehow.